src/index.js

```diff
@@ -66,15 +66,74 @@ function CordovaAuth(options) {
  */
 CordovaAuth.prototype.authorize = function (parameters, callback) {
   if (!callback || typeof callback !== 'function') {
-      throw new Error('callback not specified or is not a function');
+    throw new Error('callback not specified or is not a function');
   }
+
   var self = this;
+
   getAgent(function (err, agent) {
     if (err) {
       return callback(err);
     }
-    var authorizeUrl = self._buildAuthorizeUrlHelper(parameters);
```

Comment:
I believe @davidemerritt created these helper function to avoid duplication in the authorize_windows function, but now that we are updating the main authorize function we are losing this effort.

Personally I don't mind if the authorize_windows function is copy pasted from authorize, but the way that this code stands now we have the code duplicated + split into separate functions which makes everything confusing.

We have two options:
1. Remove the private functions and just have the code duplicated inside authorize_windows
2. Leave the private functions and make both authorize_windows and authorize use the same private functions.

So this affects both `_buildAuthorizeUrlHelper` and `_parseAuthorizeResponse`

Reply:
I think option 2 is a good path to follow to avoid adding complexity to the code and avoid this becoming a snowball. I think this way we can fix and avoid making a mess at the same time. Want to know what everyone thinks 😄